cpp/src/quantiles/tdigest/tdigest.cu

```diff
@@ -351,13 +351,17 @@ std::unique_ptr<column> percentile_approx(tdigest_column_view const& input,
   // output is a list column with each row containing percentiles.size() percentile values
   auto offsets = cudf::make_fixed_width_column(
     data_type{type_id::INT32}, input.size() + 1, mask_state::UNALLOCATED, stream, mr);
-  auto row_size_iter = thrust::make_constant_iterator(percentiles.size());
+  bool empty_input   = thrust::count_if(rmm::exec_policy(stream),
```

Comment:
```suggestion
  auto const empty_input   = thrust::count_if(rmm::exec_policy(stream),
```
Please forgive my quoting [the Good Book](https://www.goodreads.com/book/show/105125.Effective_C_): Item 3 suggests: "Use const wherever possible".
By default, consider making variables `auto const`, and then removing the `const` if they are indeed "variable".  

Also, would `all_empty_rows` be a more descriptive name? `empty_input` implies `input.size() == 0`, which isn't the case.

I think you've made the right call on using `count_if()`; I think the intention is captured better here. I'm sure `reduce()` would have worked too, but not as clear in intention. 
`thrust::all_of()` would have fit best, but [it is slow](https://github.com/NVIDIA/thrust/issues/1016).

Reply:
> I think you've made the right call on using `count_if()`...

Ah, I see that @ttnghia was here already. :p